packages/react/src/themes/teams/components/Status/statusStyles.ts

```diff
@@ -1,81 +1,64 @@
-    default:
-      return variables.defaultTextColor
-  }
-}
+    //
+    // Colors
+    //
+    [{ color: 'red' }, { backgroundColor: v.backgroundRed }],
+    [{ color: 'orange' }, { backgroundColor: v.backgroundOrange }],
+    [{ color: 'yellow' }, { backgroundColor: v.backgroundYellow }],
+    [{ color: 'green' }, { backgroundColor: v.backgroundGreen }],
+    [{ color: 'blue' }, { backgroundColor: v.backgroundBlue }],
+    [{ color: 'violet' }, { backgroundColor: v.backgroundViolet }],
 
-const sizeToPxValue: Record<SizeValue, number> = {
-  smallest: 8,
-  smaller: 8,
-  small: 8,
-  medium: 10,
-  large: 12,
-  larger: 14,
-  largest: 16,
-}
+    //
+    // States
+    //
+    [{ state: 'success' }, { backgroundColor: v.successBackgroundColor }],
+    [{ state: 'info' }, { backgroundColor: v.infoBackgroundColor }],
+    [{ state: 'warning' }, { backgroundColor: v.warningBackgroundColor }],
+    [{ state: 'error' }, { backgroundColor: v.errorBackgroundColor }],
 
-export const getSizeStyles = (sizeInPx: number, variables: StatusVariables) => {
-  const borderWidth = (variables.borderColor && variables.borderWidth) || 0
-  const sizeInRem = pxToRem(sizeInPx + 2 * borderWidth)
+    //
+    // Sizes
+    //
+    [{ size: 'smallest' }, { width: v.smallest, height: v.smallest }],
+    [{ size: 'smaller' }, { width: v.smaller, height: v.smaller }],
+    [{ size: 'small' }, { width: v.small, height: v.small }],
+    [{ size: 'medium' }, { width: v.medium, height: v.medium }],
+    [{ size: 'large' }, { width: v.large, height: v.large }],
+    [{ size: 'larger' }, { width: v.larger, height: v.larger }],
+    [{ size: 'largest' }, { width: v.largest, height: v.largest }],
+  ],
 
-  return {
-    height: sizeInRem,
-    width: sizeInRem,
-  }
-}
+  // ----------------------------------------
+  // Icon
+  // ----------------------------------------
+  icon: [
+    [{ state: 'success' }, { color: v.successTextColor }],
+    [{ state: 'info' }, { color: v.infoTextColor }],
+    [{ state: 'warning' }, { color: v.warningTextColor }],
+    [{ state: 'error' }, { color: v.errorTextColor }],
+  ],
+})
 
-const statusStyles: ComponentSlotStylesInput<StatusProps, StatusVariables> = {
-  root: ({ props: { color, size, state }, variables }): ICSSInJSStyle => {
-    return {
-      display: 'inline-flex',
-      alignItems: 'center',
-      justifyContent: 'center',
-      ...getSizeStyles(sizeToPxValue[size], variables),
-      verticalAlign: 'middle',
-      borderRadius: '9999px',
-      ...(variables.borderColor && {
-        borderColor: variables.borderColor,
-        borderWidth: pxToRem(variables.borderWidth),
-        borderStyle: 'solid',
-      }),
-      backgroundColor: color || getBackgroundColor(state, variables),
-    }
-  },
-
-  icon: ({ props: { state }, variables }): ICSSInJSStyle => ({
-    color: getTextColor(state, variables),
-  }),
-}
-
-export default statusStyles
+export default backportComponentStyle(statusStyles)
```

Comment:
`backportComponentStyle` is a helper that transforms the new proposed syntax into the previous syntax.  This just means we don't need to update the rest of our code during a migration.

If we did ship this in transition, we'd want to use this utility anywhere we have incoming styles from users.